visualization/mayavi_vis_utils.py

```diff
@@ -0,0 +1,117 @@
+
+"""
+Utilities for rendering camera frustums and 3d point clouds using Mayavi mlab.
+
+Author: John Lambert
+"""
+
+import argparse
+from typing import List
+
+import mayavi
+import numpy as np
+from colour import Color
+from gtsam import Cal3Bundler, Pose3
+from mayavi import mlab
+
+
+from gtsfm.common.view_frustum import ViewFrustum
+
+
+def draw_point_cloud_mayavi(
+    sphere_radius: float, fig: mayavi.core.scene.Scene, point_cloud: np.ndarray, rgb: np.ndarray
+) -> None:
+    """Render a point cloud as a collection of spheres, using Mayavi.
+
+    Args:
+        sphere_radius: radius of each rendered sphere.
+        fig: Mayavi figure object.
+        point_cloud: array of shape (N,3) representing 3d points.
+        rgb: uint8 array of shape (N,3) representing colors in RGB order, in the range [0,255]
+    """
+    n = point_cloud.shape[0]
+    x, y, z = point_cloud.T
+    alpha = np.ones((n, 1)).astype(np.uint8) * 255  # no transparency
+    rgba = np.hstack([rgb, alpha]).astype(np.uint8)
+
+    pts = mlab.pipeline.scalar_scatter(x, y, z)  # plot the points
+    pts.add_attribute(rgba, "colors")  # assign the colors to each point
+    pts.data.point_data.set_active_scalars("colors")
+    g = mlab.pipeline.glyph(pts)
+    g.glyph.glyph.scale_factor = sphere_radius  # set scaling for all the points
+    g.glyph.scale_mode = "data_scaling_off"  # make all the points same size
+
+
+def draw_cameras_mayavi(
+    zcwTw: Pose3, fig: mayavi.core.scene.Scene, calibrations: List[Cal3Bundler], wTi_list: List[Pose3]
+) -> None:
+    """Render camera frustums as collections of line segments, using Mayavi mlab.
+
+    Args:
+        zcwTw: transforms world points to a new world frame where the point cloud is zero-centered
+        fig: Mayavi mlab figure object.
+        calibrations: calibration object for each camera
+        wTi_list: list of camera poses for each image
```

Comment:
Same here. I honk wTi_list and calibrations should come before, as they are the most important.

Reply:
Sure. I moved `zcwTw` to the end, but still have `fig` as the first arg. let me know if this is what you had in mind

Reply:
should be fine